Tasks/AzureCloudPowerShellDeployment/Utility.ps1

```diff
@@ -0,0 +1,136 @@
+﻿function Get-SingleFile($files, $pattern)
```

Comment:
This looks like a fragile assumption that Find-VstsFiles always returns a system.array. PowerShell is intentionally vague as to when an array is used, versus arraylist, versus etc...

The way the PowerShell books recommend to coerce output from a command into an "array" (in the loose sense of the term) is to:
  `$files = @( Find-VstsFiles -LegacyPattern ... )`

Then you can check $files.Count

Wrapping a command in @( ) has the following effect:
1) When no output from the command, returns an empty array.
2) When one item is piped out of the command, returns an array with one item.
3) When multiple items are piped out of the command, returns the original array.